src/OrchardCore/OrchardCore.DisplayManagement/Implementation/DefaultHtmlDisplay.cs

```diff
@@ -97,8 +98,8 @@ public async Task<IHtmlContent> ExecuteAsync(DisplayContext context)
                     await shapeDescriptor.DisplayingAsync.InvokeAsync((action, displayContext) => action(displayContext), displayContext, _logger);
 
                     // Copy all binding sources (all templates for this shape) in order to use them as Localization scopes.
-                    shapeMetadata.BindingSources = shapeDescriptor.BindingSources.Where(x => x != null).ToList();
-                    if (!shapeMetadata.BindingSources.Any())
+                    shapeMetadata.BindingSources = shapeDescriptor.BindingSources;
```

Comment:
Can you check that no `null` values can be set as `Bindingsources` in any way?

Reply:
in the [ShapeAlterationBuilder.cs](https://github.com/OrchardCMS/OrchardCore/pull/15652/files#diff-2d7bc4e830df8b3f18c704125fe613f0489c8acf3e080a03f348ca7805f6f72a) which is why I added a null check. and if needed, we can add `.Where(x => x  != null)` in the `ShapeDescriptorIndex` class instead so it is not evaluated on every call.